torch/csrc/jit/tracer.cpp

```diff
@@ -191,6 +191,45 @@ Value* getNestedOutputTrace(
   }
 }
 
+inline TypePtr typeForTraceInput(const IValue& input) {
+  if (input.isGenericDict()) {
```

Comment:
Looks great!! Not sure if this has been already checked but just to be safe throw an error for GenericList?

& throw an error if the GenericDict is empty?